modules/calib3d/src/fisheye.cpp

```diff
@@ -161,49 +160,40 @@ void cv::fisheye::projectPoints(InputArray objectPoints, OutputArray imagePoints
             Matx33d dYdT_data = Matx33d::eye();
             const Vec3d *dYdT = (Vec3d*)dYdT_data.val;
 
-            //Vec2d x(Y[0]/Y[2], Y[1]/Y[2]);
-            Vec3d dxdom[2];
-            dxdom[0] = (1.0/Y[2]) * dYdom[0] - x[0]/Y[2] * dYdom[2];
-            dxdom[1] = (1.0/Y[2]) * dYdom[1] - x[1]/Y[2] * dYdom[2];
-
-            Vec3d dxdT[2];
-            dxdT[0]  = (1.0/Y[2]) * dYdT[0] - x[0]/Y[2] * dYdT[2];
-            dxdT[1]  = (1.0/Y[2]) * dYdT[1] - x[1]/Y[2] * dYdT[2];
-
-            //double r2 = x.dot(x);
-            Vec3d dr2dom = 2 * x[0] * dxdom[0] + 2 * x[1] * dxdom[1];
-            Vec3d dr2dT  = 2 * x[0] *  dxdT[0] + 2 * x[1] *  dxdT[1];
-
-            //double r = std::sqrt(r2);
-            double drdr2 = r > 1e-8 ? 1.0/(2*r) : 1;
-            Vec3d drdom = drdr2 * dr2dom;
-            Vec3d drdT  = drdr2 * dr2dT;
-
-            // Angle of the incoming ray:
-            //double theta = atan(r);
-            double dthetadr = 1.0/(1+r2);
-            Vec3d dthetadom = dthetadr * drdom;
-            Vec3d dthetadT  = dthetadr *  drdT;
-
-            //double theta_d = theta + k[0]*theta3 + k[1]*theta5 + k[2]*theta7 + k[3]*theta9;
-            double dtheta_ddtheta = 1 + 3*k[0]*theta2 + 5*k[1]*theta4 + 7*k[2]*theta6 + 9*k[3]*theta8;
-            Vec3d dtheta_ddom = dtheta_ddtheta * dthetadom;
-            Vec3d dtheta_ddT  = dtheta_ddtheta * dthetadT;
-            Vec4d dtheta_ddk  = Vec4d(theta3, theta5, theta7, theta9);
-
-            //double inv_r = r > 1e-8 ? 1.0/r : 1;
-            //double cdist = r > 1e-8 ? theta_d / r : 1;
-            Vec3d dcdistdom = inv_r * (dtheta_ddom - cdist*drdom);
-            Vec3d dcdistdT  = inv_r * (dtheta_ddT  - cdist*drdT);
-            Vec4d dcdistdk  = inv_r *  dtheta_ddk;
-
-            //Vec2d xd1 = x * cdist;
+            // double r2 = x.dot(x);
+            Vec3d dr2dom = 2 * (x[0] * dYdom[0] + x[1] * dYdom[1]);
+            Vec3d dr2dT  = 2 * (x[0] * dYdT[0]  + x[1] * dYdT[1]);
+            // double z = Y[2];
+            const Vec3d& dzdom = dYdom[2];
+            const Vec3d& dzdT  = dYdT[2];
+
+            // double theta_ = r<1e-8*z ? 1/z - r2/(3*z*z*z) : atan2(r, z)/r;
+            double dtheta_dr2 = r<1e-8*z ? -1/(3*z3) : 1/(2*r2)*(z/(r2+z2) - theta_);
```

Comment:
N.B.: The small `r` derivation must actually be `+1/(6*z3)` -- I'll update this PR if I get any positive response ...